packages/d3fc-webgl/src/series/glLine.js

```diff
@@ -140,6 +131,39 @@ export default () => {
         return draw;
     };
 
+    draw.defined = (...args) => {
+        const builder = program.buffers().attribute(definedAttrib);
+
+        const definedArray = new Float32Array(args[0].length * 4);
+        
+        const length = definedArray.length;
```

Comment:
Without knowing exactly what this is up to, this seems very confusing. I wonder if it think it might be clearer if it was done as a loop of the `args[0]` array (please give this a name!) instead?

Reply:
I've refactored this and used a loop of the args[0] array, hopefully it should be a bit clearer now. Let me know if it is still confusing